demo/text_matching/text_matching.py

```diff
@@ -0,0 +1,97 @@
+parser.add_argument("--num_epoch", type=int, default=3, help="Number of epoches for fine-tuning.")
+parser.add_argument("--use_gpu", type=ast.literal_eval, default=True, help="Whether use GPU for fine-tuning, input should be True or False")
+parser.add_argument("--learning_rate", type=float, default=5e-5, help="Learning rate used to train with warmup.")
+parser.add_argument("--checkpoint_dir", type=str, default=None, help="Directory to model checkpoint")
+parser.add_argument("--max_seq_len", type=int, default=512, help="Number of words of the longest seqence.")
+parser.add_argument("--batch_size", type=int, default=32, help="Total examples' number in batch for training.")
+parser.add_argument("--network", type=str, default=None, help="Pre-defined network which was connected after module.")
+parser.add_argument("--use_data_parallel", type=ast.literal_eval, default=False, help="Whether use data parallel.")
+parser.add_argument("--is_pair_wise", type=ast.literal_eval, default=False, help="Whether use data parallel.")
+args = parser.parse_args()
+# yapf: enable.
+
+if __name__ == '__main__':
+
+    # Load Paddlehub ERNIE pretrained model
+    module = hub.Module(name="ernie")
+
+    # Pair wise task needs: query, title_left, right_title (3 data)
+    # Point wise task needs: query, title_left (2 data)
+    if args.is_pair_wise:
+        num_data = 3
+    else:
+        num_data = 2
+    inputs, outputs, program = module.context(
+        trainable=True, max_seq_len=args.max_seq_len, num_data=num_data)
+
+    # Tokenizer tokenizes the text data and encodes the data as model needed.
+    # If you use transformer modules (ernie, bert, roberta and so on), tokenizer should be hub.BertTokenizer.
+    # else tokenizer should be hub.CustomTokenizer.
+    tokenizer = hub.BertTokenizer(
+        vocab_file=module.get_vocab_path(), tokenize_chinese_chars=True)
+
+    # Load dataset
+    if args.is_pair_wise:
+        dataset = hub.dataset.DuEL(
+            tokenizer=tokenizer, max_seq_len=args.max_seq_len)
+    else:
+        dataset = hub.dataset.LCQMC(
+            tokenizer=tokenizer, max_seq_len=args.max_seq_len)
+
+    # Construct transfer learning network
+    # Use token-level output.
+    query = outputs["sequence_output"]
+    left = outputs['sequence_output_2']
+    right = outputs['sequence_output_3']
+
+    # Select fine-tune strategy
+    strategy = hub.DefaultStrategy(
+        optimizer_name="sgd", learning_rate=args.learning_rate)
+
+    # Setup RunConfig for PaddleHub Fine-tune API
+    config = hub.RunConfig(
+        eval_interval=300,
+        use_data_parallel=args.use_data_parallel,
+        use_cuda=args.use_gpu,
+        num_epoch=args.num_epoch,
+        batch_size=args.batch_size,
+        checkpoint_dir=args.checkpoint_dir,
+        strategy=strategy)
+
+    # Define a text matching task by PaddleHub's API
+    # network choice: bow, cnn, gru, lstm (PaddleHub pre-defined network)
+    matching_task = hub.TextMatchingTask(
```

Comment:
How about separate TextMatchingTask into 2 task
`PairwiseTextMatchingTask` and `PointwiseTextMatchingTask`
so the args should change to  
`query_feature`, `title_feature`
`query_feature`, `left_title_feature`, `right_title_feature`
I think that would be clearer and easier to understand compare to the current API design